compiler/rustc_span/src/source_map.rs

```diff
@@ -632,10 +632,11 @@ impl SourceMap {
     pub fn span_to_margin(&self, sp: Span) -> Option<usize> {
         match self.span_to_prev_source(sp) {
             Err(_) => None,
-            Ok(source) => source
-                .split('\n')
-                .last()
-                .map(|last_line| last_line.len() - last_line.trim_start().len()),
+            Ok(source) => {
+                let last_line = source.rsplit_once('\n').unwrap_or(("", &source)).1;
+
+                Some(last_line.len() - last_line.trim_start().len())
```

Comment:
This now returns `Some` instead of `None` when there is no `\n`. Was this an intentional change?

Reply:
If there any string, even without `\n`, it won't return `None`:
```rust
assert_eq!(Some("abcdef"), "abcdef".split('\n').last());
```

Reply:
Ah, of course, apologies.